sql/src/main/java/org/apache/druid/sql/http/SqlResource.java

```diff
@@ -196,7 +197,7 @@ public Response doPost(
       endLifecycle(sqlQueryId, lifecycle, timeout, remoteAddr, -1);
       return buildNonOkResponse(QueryTimeoutException.STATUS_CODE, timeout, sqlQueryId);
     }
-    catch (SqlPlanningException | ResourceLimitExceededException e) {
+    catch (SqlPlanningException | ResourceLimitExceededException | BadQueryContextException e) {
```

Comment:
Would it make sense to just catch `BadQueryException` here instead of all its subclasses?

Reply:
The only class left out is `BadJsonQueryException`, I don't see any reason why it can't be caught here too.